.github/workflows/ci.yml

```diff
@@ -5,6 +5,11 @@ on:
     branches-ignore:
       - 'main'
 
+# Allow one instance of this workflow per pull request, and cancel older runs when new changes are pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
```

Comment:
trivial: this is slightly different than what we have in [aws-crt-java](https://github.com/awslabs/aws-crt-cpp/blob/281a7caff7e10f68a5422d8fca8acf0b48e4215f/.github/workflows/ci.yml#L22) and aws-crt-cpp, which do this:
> group: ${{ github.workflow }}-${{ github.ref }}

which I got from these docs:
https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs